src/components/timeline/index.js

```diff
@@ -72,29 +72,84 @@ type DispatchProps = {|
 
 type Props = ConnectedProps<OwnProps, StateProps, DispatchProps>;
 
-class Timeline extends React.PureComponent<Props> {
+class TimelineSettingsGraphType extends React.PureComponent<{|
+  +timelineType: TimelineType,
+  +changeTimelineType: typeof changeTimelineType,
+|}> {
   _changeToCategories = () => this.props.changeTimelineType('category');
   _changeToStacks = () => this.props.changeTimelineType('stack');
```

Comment:
super optional: you can inline these functions below. This won't trigger renders because they are properties do a DOM element.
I'd find it better because then we clearly see that the argument (eg: `'category'`) matches the check above (eg: `'timelineType === 'category'`).